src/Core/Coverage.idr

```diff
@@ -24,20 +24,20 @@ conflictMatch : {vars : _} -> List (Name, Term vars) -> Bool
 conflictMatch [] = False
 conflictMatch ((x, tm) :: ms) = conflictArgs x tm ms || conflictMatch ms
   where
-    clash : Term vars -> Term vars -> Bool
+    clash : Term vars -> Term vars -> Maybe Bool
```

Comment:
Can you please document the meaning of this `Maybe`? Or should we even define
e.g. `data ClashingDecision = Distinct | Same | Incomparable`
with the idea that we are only checking argument if they're `Same`?

Reply:
I was on the fence about whether to use an enum or not.  Coming back to this a week later and having to get my bearings again, I think it would be clearer if I did that.